psw/ae/aesm_service/source/aesm/application/main.cpp

```diff
@@ -73,8 +73,12 @@ void signal_handler(int sig)
     }
 }
 
-int main() {
-    if(daemon(0, 0) < 0)
+int main(int argc, char *argv[]) {
+    // The only command line option that is supported is --no-daemon,
+    // so for now we can hardcode a check against argv[1], which will
+    // be NULL if no argument is provided.
+    bool noDaemon = (strcmp(argv[1], "--no-daemon") == 0);
```

Comment:
Using `strncmp()` is functionally incorrect and unnecessary.  Excepting a problem in the compiler or libc, in which case we have bigger issues to worry about, `--no-daemon` and `argv[1]` are guaranteed to be null terminated.  Using `strncmp()` will result in false positives, e.g. `strncmp()` will return 0 for invocations like `aesm_service --no-daemon=false` and `aesm_service --no-daemonization`.